src/main/java/com/practise/tdd/PerfectNumberValidator.java

```diff
@@ -1,10 +1,48 @@
 package com.practise.tdd;
 
+import java.util.ArrayList;
+import java.util.List;
 
 public class PerfectNumberValidator {
 
 	public boolean isPerfectNumber(int number) {
+		List<Integer> divisorsOfNumber = getAllDivisors(number);
+		
+		if(number!= 1 && sumOfDivisorsEqualsNumber(divisorsOfNumber, number))
+			return true;
+		
 		return false;
 	}
 	
+	private List<Integer> getAllDivisors(int dividend) {
+		List<Integer> validDivisors = new ArrayList<>();
+		validDivisors.add(new Integer(1));
+		
+		int remainder = dividend;
+		while(remainder > 1) {
+			int[] primeNumberDivisors = {2,3,5,7};
+			
```

Comment:
This code was initially used because I was trying to find divisors using Prime Factorization algorithm. This is also removed in final solution.